File: flutter_theoplayer_sdk_android/android/src/main/kotlin/com/theoplayer/theoplayer/TheoplayerPlugin.kt
package com.theoplayer.theoplayer

import io.flutter.embedding.engine.plugins.FlutterPlugin

class TheoplayerPlugin : FlutterPlugin {

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        flutterPluginBinding.platformViewRegistry.registerViewFactory(
            "com.theoplayer/theoplayer-view-native",
            THEOplayerViewNativeFactory(flutterPluginBinding.binaryMessenger)
        )
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {

    }
}

File: flutter_theoplayer_sdk/flutter_theoplayer_sdk_ios/ios/Classes/TheoplayerPlugin.swift
import Flutter
import UIKit

public class TheoplayerPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "theoplayer", binaryMessenger: registrar.messenger())
    let instance = TheoplayerPlugin()
    
    let messenger = registrar.messenger()
      
    PlatformActivityService.ensureInitialized(messenger: messenger)

    registrar.addMethodCallDelegate(instance, channel: channel)
    registrar.register(THEOplayerNativeViewFactory(messenger: messenger), withId: "com.theoplayer/theoplayer-view-native")
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
